merlin/schema/schema.py

```diff
@@ -40,6 +40,11 @@ class ColumnSchema:
     is_ragged: bool = False
 
     def __post_init__(self):
+        """Standardize tags and dtypes on initialization
+
+        Raises:
+            TypeError: If provided dtype can't be cast to a numpy dtype
```

Comment:
Up to you: I'd spell out "cannot" and I'd include "the" to avoid telegraphic language:

> TypeError: If the provided dtype cannot be cast to a numpy dtype